### PR TITLE
refactor work-manager addon

### PR DESCRIPTION
**Pull request by zhiweiyin318**
story: https://github.com/stolostron/backlog/issues/19280
Signed-off-by: Zhiwei Yin <zyin@redhat.com>

### CODE REVIEW

File: cmd/controller/app/server.go [hunk 1]
```diff
@@ -134,7 +138,23 @@ func Run(o *options.ControllerRunOptions, ctx context.Context) error {
 		return err
 	}
 	if o.EnableAddonDeploy {
-		addonMgr.AddAgent(addon.NewAgent(kubeClient, "work-manager", o.AddonImage, o.AddonInstallNamespace))
+		addonName := "work-manager"
```
Comment (qiujian16) [inline cmd/controller/app/server.go:141]: const?

File: pkg/addon/addon.go [hunk 2]
```diff
@@ -3,129 +3,97 @@ package addon
...
-				// TODO add more csr check here.
-				return true
-			},
-			PermissionConfig: func(cluster *clusterv1.ManagedCluster, addon *addonapiv1alpha1.ManagedClusterAddOn) error {
-				return f.createOrUpdateRoleBinding(cluster.Name)
-			},
+func NewRegistrationOption(kubeClient kubernetes.Interface, addonName string) *agent.RegistrationOption {
+	return &agent.RegistrationOption{
+		CSRConfigurations: agent.KubeClientSignerConfigurations(addonName, addonName),
+		CSRApproveCheck: func(
```
Comment (qiujian16) [inline pkg/addon/addon.go:74]: i think you can use the approve helper in addon-framework

Comment (zhiweiyin318) [reply pkg/addon/addon.go]: done

File: cmd/controller/app/server.go (hunk 1 above)
Comment (zhiweiyin318) [reply cmd/controller/app/server.go]: done

**Pull request merged**